pkg/kubelet/kuberuntime/kuberuntime_manager_test.go

```diff
@@ -920,6 +920,60 @@ func TestComputePodActions(t *testing.T) {
 	}
 }
 
+func makeBasePodAndStatusWithSidecar() (*v1.Pod, *kubecontainer.PodStatus) {
+	pod, status := makeBasePodAndStatus()
+	sidecarType := v1.LifecycleTypeSidecar
+	pod.Spec.Containers[1].Lifecycle = &v1.Lifecycle{Type: &sidecarType}
+	status.ContainerStatuses[1].Hash = kubecontainer.HashContainer(&pod.Spec.Containers[1])
+
+	return pod, status
+}
+
+func TestComputePodActionsWithSidecar(t *testing.T) {
+	_, _, m, err := createTestRuntimeManager()
+	require.NoError(t, err)
+
+	// Creating a pair reference pod and status for the test cases to refer
+	// the specific fields.
+	basePod, baseStatus := makeBasePodAndStatusWithSidecar()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SidecarLifecycle, true)()
+	for desc, test := range map[string]struct {
```

Comment:
Can we add more test cases here to cover various cases? roughly combinations of the followings:
- standard container is running / created / nil status
- different restart policies
- container status has nil / created / unknown
- container exit code 0 / non 0

Reply:
yeah I want to add more tests, those are all great suggestions thanks

Reply:
Glad to see the extensive test coverage here! To be honest, the `computePodActions` method is quite long and very tough to follow. So I think good test coverage here is essential.